jobspec/parse.go

```diff
@@ -452,6 +462,81 @@ func parseTasks(result *[]*structs.Task, list *ast.ObjectList) error {
 	return nil
 }
 
+func parseServices(jobName string, taskGroupName string, task *structs.Task, serviceObjs *ast.ObjectList) error {
+	task.Services = make([]structs.Service, len(serviceObjs.Items))
+	var defaultServiceName bool
+	for idx, o := range serviceObjs.Items {
+		var service structs.Service
+		var m map[string]interface{}
+		if err := hcl.DecodeObject(&m, o.Val); err != nil {
+			return err
+		}
+
+		delete(m, "check")
+
+		if err := mapstructure.WeakDecode(m, &service); err != nil {
+			return err
+		}
+
+		if defaultServiceName && service.Name == "" {
+			return fmt.Errorf("More than one service block is declared, please name each service explicitly")
```

Comment:
This has the wrong message.

`return fmt.Errorf("Only one service block may omit name.")